site/app/templates/submission/homework/CurrentVersionResults.twig

```diff
@@ -16,11 +16,18 @@
                         <span>
                             {{ file.relative_name }} ({{ (file.size / 1024) | number_format(2) | default(-1) }}kb)
                         </span>
-                    {# download icon if student can download files #}
-                    {% if student_download %}
-                        <button class = 'btn btn-primary key_to_click' onclick='downloadFile("{{ file.path|url_encode }}", "submissions")' aria-label="Download {{file.relative_name}}"> Download
-                            <i class="fas fa-download" title="Download the file"></i></button>
-                    {% endif %}
+                        {# view and download icons if student is permitted to access files #}
+                        {% if student_download %}
+                            <div>
+                                {% set extension = file.relative_name|split('.')|last|lower %}
+                                {% if '.' ~ extension in ['.pdf', '.jpg', '.jpeg', '.c', '.cpp', '.s', '.twig', '.py', '.java', '.png', '.txt', '.h', '.html', '.php', '.js', '.sql', '.sh', '.md', '.csv', '.salsa', '.erl', '.oz', '.pl', '.hs', '.gif'] %}
```

Comment:
You shouldn't need to concatenate the '.' to extension if you remove from '.' each item in the list of acceptable extensions.

Reply:
That code is copied over from `CourseMaterials.twig`; I didn't change anything to keep it consistent.
If we want to refactor it we should do it in both places.